### PR TITLE
Adding initial infra for supporting ECS-based jobs

### PR DESCRIPTION
**Pull request by blawney**

### CODE REVIEW

File: deployment-aws/terraform/ecs.tf [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "ecs_s3_access" {
+  name   = "AllowECSAccessToStorageBucket"
```
Comment (hackdna) [inline deployment-aws/terraform/ecs.tf]: Should this name have a `${local.common_tags.Name}` prefix for easier identification?

File: deployment-aws/terraform/ecs.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          ]
+        }
+      ]
+    }
+  )
+}
+
+
+resource "aws_iam_role_policy" "efs_ap" {
+    name   = "EFSAccessPointForECS"
```
Comment (hackdna) [inline deployment-aws/terraform/ecs.tf]: Should this name have a `${local.common_tags.Name}` prefix for easier identification?

File: deployment-aws/terraform/efs.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  # implicit with AWS but Terraform requires this to be explicit
+  egress {
+    description      = "Allow all egress"
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
```
Comment (hackdna) [inline deployment-aws/terraform/efs.tf:27]: Is this egress block actually needed here?

File: deployment-aws/terraform/network.tf [hunk 4]
```diff
@@ -151,3 +152,153 @@ resource "aws_security_group" "api_server" {
...
+  private_dns_enabled = true
+}
+
+# Although we don't explicitly create a route table for the 
+# private subnets, it is created automatically and associated
+# with both
+# data "aws_route_table" "private_route_table" {
+#   vpc_id    = aws_vpc.main.id
+#   subnet_id = aws_subnet.private_a.id
+# }
```
Comment (hackdna) [inline deployment-aws/terraform/network.tf]: Commented out code should be deleted.

File: deployment-aws/terraform/network.tf [hunk 5]
```diff
@@ -151,3 +152,153 @@ resource "aws_security_group" "api_server" {
...
+  vpc_id      = aws_vpc.main.id
+  ingress {
+    description      = "Allow ingress from ECS sg"
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    security_groups  = [aws_security_group.ecs_instance_security_group.id]
+  }
+  # implicit with AWS but Terraform requires this to be explicit
+  egress {
```
Comment (hackdna) [inline deployment-aws/terraform/network.tf]: Is this actually needed here?

**Review by hackdna — CHANGES_REQUESTED** [review]

File: deployment-aws/terraform/ecs.tf (hunk 1 above)
Comment (blawney) [reply deployment-aws/terraform/ecs.tf]: yes, done

File: deployment-aws/terraform/ecs.tf (hunk 2 above)
Comment (blawney) [reply deployment-aws/terraform/ecs.tf]: yes, done

File: deployment-aws/terraform/efs.tf (hunk 3 above)
Comment (blawney) [reply deployment-aws/terraform/efs.tf:27]: confirmed it's not on a standalone stack. removed.

File: deployment-aws/terraform/network.tf (hunk 5 above)
Comment (blawney) [reply deployment-aws/terraform/network.tf]: nope, confirmed it works elsewhere

File: deployment-aws/terraform/network.tf (hunk 4 above)
Comment (blawney) [reply deployment-aws/terraform/network.tf]: I had left that as a reminder about the `data` resource in case we were able to figure that out.

**Review by hackdna — APPROVED** [review]

File: deployment-aws/terraform/network.tf [hunk 6]
```diff
@@ -151,3 +158,150 @@ resource "aws_security_group" "api_server" {
...
+
+# Although we don't explicitly create a route table for the 
+# private subnets, it is created automatically and associated
+# with both
+# data "aws_route_table" "private_route_table" {
+#   vpc_id    = aws_vpc.main.id
+#   subnet_id = aws_subnet.private_a.id
+# }
+
+resource "aws_route_table" "private_route_table" {
```
Comment (hackdna) [inline deployment-aws/terraform/network.tf]: No need to include the resource type in the name:
```
resource "aws_route_table" "private"
```
Then you can refer to it without redundancy:
```
aws_route_table.private
```

**Review by hackdna — APPROVED** [review]

**Pull request merged**